inference-engine/src/vpu/common/src/ngraph/operations/static_shape_broadcast.cpp

```diff
@@ -19,30 +19,32 @@ StaticShapeBroadcast::StaticShapeBroadcast(const Output<Node>& arg,
                                            const Output<Node>& targetShape,
                                            const Output<Node>& axesMapping,
                                            const ngraph::op::BroadcastModeSpec& broadcastSpec)
-        : ::ngraph::op::v3::Broadcast{arg, targetShape, axesMapping, broadcastSpec} {
+        : ::ngraph::op::v3::Broadcast{arg, targetShape, axesMapping, broadcastSpec},
+          m_evaluatedOutputShape{PartialShape::dynamic()} {
     constructor_validate_and_infer_types();
 }
 
 StaticShapeBroadcast::StaticShapeBroadcast(const Output<Node>& arg,
                                            const Output<Node>& targetShape,
                                            const ngraph::op::BroadcastModeSpec& broadcastSpec)
-        : ::ngraph::op::v3::Broadcast{arg, targetShape, broadcastSpec} {
+        : ::ngraph::op::v3::Broadcast{arg, targetShape, broadcastSpec},
+          m_evaluatedOutputShape{PartialShape::dynamic()} {
     constructor_validate_and_infer_types();
 }
 
 void StaticShapeBroadcast::validate_and_infer_types() {
-    if (get_output_partial_shape(0).is_static()) {
-        return;
+    auto& outputShape = m_evaluatedOutputShape;
+    if (outputShape.is_dynamic()) {
+        ::ngraph::op::v3::Broadcast::validate_and_infer_types();
```

Comment:
Do I understand correctly, that you need to call `::ngraph::op::v3::Broadcast::validate_and_infer_types` always and then if evaluated output shape is not dynamic reset output shape with evaluated one?

Reply:
Yes, you're right.

Reply:
If so why you call `::ngraph::op::v3::Broadcast::validate_and_infer_types` only if `outputShape` is dynamic?

Reply:
Well, maybe I still misunderstood your question. `m_evaluatedOutputShape` is initialized as `PartialShape::dynamic()`, so we go into this `if` only if we are running the validation very first time. If we were able to calculate static output shape, we save it in `m_evaluatedOutputShape`, otherwise we fail. The difference between this and previous approach is that I cache evaluated output shape and run `set_output_type` each time `validate_and_infer_types` is called for StaticShape* operations.